services/director-v2/tests/unit/test_modules_dynamic_sidecar_client_api_base_client.py

```diff
@@ -0,0 +1,123 @@
+# pylint:disable=redefined-outer-name
+
+import pytest
+from _pytest.logging import LogCaptureFixture
+from httpx import ConnectError, Response, codes
+from pydantic import AnyHttpUrl, parse_obj_as
+from respx import MockRouter
+from simcore_service_director_v2.modules.dynamic_sidecar.api_client._base import (
+    BaseHThinClient,
+    expect_status,
+    retry_on_errors,
+)
+from simcore_service_director_v2.modules.dynamic_sidecar.api_client._errors import (
+    ClientTransportError,
+    UnexpectedStatusError,
+    WrongReturnType,
+)
+
+pytestmark = pytest.mark.asyncio
+
+# UTILS
+
+
+class TestThickClient(BaseHThinClient):
+    @retry_on_errors
+    async def get_provided_url(self, provided_url: str) -> Response:
+        return await self._client.get(provided_url)
+
+    @retry_on_errors
+    async def get_retry_for_status(self) -> Response:
+        return await self._client.get("http://some-test.url")
+
+
+# FIXTURES
+
+
+@pytest.fixture
+def thick_client() -> TestThickClient:
+    return TestThickClient(request_max_retries=1)
+
+
+@pytest.fixture
+def test_url() -> AnyHttpUrl:
+    return parse_obj_as(AnyHttpUrl, "http://some-host:8008")
```

Comment:
MINOR: use faker to create randmo urls

Reply:
I've looked into it and will not use it for two reasons:
- it does not have support to include the port
- but most importantly it can create existing addresses which will make tests fail

I've opted for using the same `http://missing-host:1111` across all test mocks.